dasharo-compatibility/cpu-cores-count.robot

```diff
@@ -0,0 +1,236 @@
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Detect Or Install Package    util-linux    
+    ${out}=    Execute Command In Terminal    lscpu | grep -v "NUMA" | grep "CPU(s):"
+    ${cores}=    Get Amount Of E cores
+    ${cores}=    Evaluate    ${cores} + 2
+    ${cores}=    Convert To String    ${cores}
+    Should Contain    ${out}    ${cores}
+
+CCC004.001 CPU P-cores only one active, CPU E-cores disabled, Hyper-Threading disabled (Ubuntu)
+    [Documentation]    Check if the correct amount of cores is active when
+    ...    all of E-cores are disabled, all P-cores are disabled and 
+    ...    Hyper-Threading is disabled.
+    Set UEFI Option    HyperThreading    ${FALSE}
+    Set UEFI Option    ActiveECores    0
+    Set UEFI Option    ActivePCores    1
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Detect Or Install Package    util-linux    
+    ${out}=    Execute Command In Terminal    lscpu | grep -v "NUMA" | grep "CPU(s):"    
+    Should Contain    ${out}    1
+
+CCC004.002 CPU P-cores only one active, CPU E-cores disabled, Hyper-Threading enabled (Ubuntu)
+    [Documentation]    Check if the correct amount of cores is active when
+    ...    all of E-cores are disabled, all P-cores are disabled and 
+    ...    Hyper-Threading is disabled.
+    Set UEFI Option    HyperThreading    ${TRUE}
+    Set UEFI Option    ActiveECores    0
+    Set UEFI Option    ActivePCores    1
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Detect Or Install Package    util-linux    
+    ${out}=    Execute Command In Terminal    lscpu | grep -v "NUMA" | grep "CPU(s):"    
+    Should Contain    ${out}    2
+
+CCC005.001 CPU P-cores all active, CPU E-cores all active, Hyper-Threading enabled (Ubuntu)
+    [Documentation]    Check if the correct amount of cores is active when
+    ...    all of E-cores are disabled, all P-cores are disabled and 
+    ...    Hyper-Threading is disabled.
+    Set UEFI Option    HyperThreading    ${TRUE}
+    Set UEFI Option    ActiveECores    All active
+    Set UEFI Option    ActivePCores    All active
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Detect Or Install Package    util-linux    
+    ${out}=    Execute Command In Terminal    lscpu | grep -v "NUMA" | grep "CPU(s):"    
+    Should Contain    ${out}    ${DEF_THREADS_TOTAL}
+
+*** Keywords ***
+Get Threads Per Core
+    [Arguments]
+    ${out}=    Execute Command In Terminal    lscpu | grep "Thread(s) per core: "
+    ${threads}=    Get Last Word From String    ${out}
+    IF    '${threads[:1]}' == '1' 
+        ${count}=    Set Variable    1
+    ELSE IF    '${threads[:1]}' == '2'
+        ${count}=    Set Variable    2 
+    ELSE
+        Fail    Hyper-Threading status could not be established.
+    END
+    RETURN     ${count}
+
+Get Last Word From String
```

Comment:
The proposed keyword seems too complex.

Would something like this (or similar) work?

```
    ${words}=  Split String  ${string}  ${SPACE}
    ${last_word}=  Get From List  ${words}  -1
```

Alternatively, if you want to use python anyway, you can create a Python function in a .py file. But RF should be enough here.

Reply:
Changed in: 58b37a0a3e56b4f758ad352d5db6046131e7ece2